File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    // Set the initial amount of coins to 0
    int coins = 0;
    // Program so that the inserted change amount is not negative
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);
    // Multiply dollar amount by 100
    change = change * 100;
    // Round change to prevent errors
    int r = round(change);
    // Go down the list of quarters, dimes, nickels, and pennies, and ensure that the code tries to convert the change to coins in that order of preference
    // Ensure that the amount of coins increases by 1 whenever the change is converted
    while (r >= 25)
    {
        r = r - 25;
        coins = coins + 1;
    }
    while (r >= 10)
    {
        r = r - 10;
        coins = coins + 1;
    }
    while (r >= 5)
    {
        r = r - 5;
        coins = coins + 1;
    }
    while (r >= 1)
    {
        r = r - 1;
        coins = coins + 1;
    }
    // Have code display amount of coins
    printf("%i\n", coins);
}

File: cash.py
from cs50 import get_float


def main():
    # Prompt user for float value for change until nonnegative value is given
    while True:
        change = get_float('Change owed: ')
        if change > 0:
            break

    # Convert change (in dollars) to integer amount of cents
    cents = round(change * 100)
    # Initialize counter for coins
    coins = 0
    # Add as many 25-cent groups (quarters) as possible from cents into counter
    coins += cents // 25
    # Cents remaining is the remainder from the amount of cents before being divided by 25
    cents = cents % 25
    # Add as many 10-cent groups (dimes) as possible from cents into counter
    coins += cents // 10
    # Cents remaining is the remainder from the amount of cents before being divided by 10
    cents = cents % 10
    # Add as many 5-cent groups (nickels) as possible from cents into counter
    coins += cents // 5
    # Cents remaining is the remainder from the amount of cents before being divided by 5
    cents = cents % 5
    # Add remaining cents (pennies) to counter
    coins += cents
    # Print counter
    print(coins)


if __name__ == '__main__':
    main()
